src/Tribe/Repositories/Attendee/RSVP.php

```diff
@@ -104,4 +105,123 @@ public function checked_in_keys() {
 		return $this->limit_list( $this->key_name, parent::checked_in_keys() );
 	}
 
+	/**
+	 * Update Additional data after creation of attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post                       $attendee Attendee Object.
+	 * @param Tribe__Tickets__Ticket_Object $ticket Ticket Object.
+	 * @param array                         $attendee_data Array of attendee data.
+	 */
+	public function update_additional_data( $attendee, $ticket, $attendee_data ) {
+
+		$attendee_data = $this->format_attendee_data( $attendee, $ticket, $attendee_data );
+
+		$query = $this->by( 'id', $attendee->ID );
+
+		try {
+			$query->set_args( $attendee_data );
+		} catch ( Tribe__Repository__Usage_Error $e ) {
+			do_action( 'tribe_log', 'error', __CLASS__, [ 'message' => $e->getMessage() ] );
+			return;
+		} finally {
+			$query->save();
+
+			$this->trigger_actions( $attendee, $ticket, $attendee_data );
+		}
```

Comment:
Let's prevent `finally` from running the save etc if there was a usage error `$query`.

```suggestion
		}

		$query->save();

		$this->trigger_actions( $attendee, $ticket, $attendee_data );
```